### ISSUE TITLE
[Feature] Add support for Sourmash

### ISSUE THREAD

**Issue by Midnighter** [taxprofiler/taxpasta#114]
I think sourmash is an interesting tool, as it is so fast in scanning vast libraries of genomes. We should add support for its output.

**Comment by chrisgulvik** [comment]
@Midnighter Any progress on this? I'm also interested in having it for evaluation.

According to the docs `sourmash tax` is the recommended approach (not `sourmash lca` anymore) [link](https://sourmash.readthedocs.io/en/latest/classifying-signatures.html#sourmash-tax-vs-sourmash-lca). I don't have example commands to use, but there's a fairly recent nf wf that might be helpful if it's the cmds themselves is what's slowing you down [here](https://github.com/lephuocquyen/Sourmash-Taxonomy-Profiling/blob/main/workflows/sourmash.nf). It looks like the main steps are:
1. sketch the input to form a sig (`sourmash sketch`) [here](https://github.com/lephuocquyen/Sourmash-Taxonomy-Profiling/blob/main/modules/sketch/main.nf#L28-L31)
2. search the sig against a db (`sourmash gather`) [here](https://github.com/lephuocquyen/Sourmash-Taxonomy-Profiling/blob/main/modules/gather/main.nf#L39-L47)
3. summarize results by lineage (`sourmash tax metagenome`) [here](https://github.com/lephuocquyen/Sourmash-Taxonomy-Profiling/blob/main/modules/taxmetagenome/main.nf#L38-L46)
4. annotate results (`sourmash tax annotate`) [here](https://github.com/lephuocquyen/Sourmash-Taxonomy-Profiling/blob/main/modules/taxannotate/main.nf#L24-L29)

where steps 3 and 4 could occur in parallel.

The bioconda is up-to-date [here](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/sourmash/meta.yaml), databases are well-described [here](https://sourmash.readthedocs.io/en/latest/databases.html#gtdb-r07-rs207-dna-databases), and the software itself is very well maintained by @ctb et al. for almost a decade now. Also including him to give an opportunity to suggest alternative cmds for generalized classification, in case the above steps are less than ideal.

**Comment by ctb** [comment]
I STAND READY

😆

**Comment by ctb** [comment]
can anyone give an example of one or two use cases so I can read the docs a bit with that in mind? would the `standardize` command be a good place to start?

might be fun to add [sylph](https://github.com/bluenote-1577/sylph) support as well, since people are liking that a lot (I'm not a maintainer - that would be @bluenote-1577)

**Comment by Midnighter** [comment]
Thank you for your interest @chrisgulvik 🙂. As this is taxpasta and not the taxprofiler pipeline, the exact commands actually don't matter in this context. The only thing required from a technical perspective are examples of a few profiles created with sourmash and maybe a clear understanding what variation in terms of column output is possible/desirable/supportable.

The major impediment is my time really, as I have moved into a different job, and taxpasta is now essentially a hobby project among (several) others. We have a fairly decent guide for how to add support for new types of profiles (https://taxpasta.readthedocs.io/en/latest/contributing/supporting_new_profiler/), so if you want to give it a shot, I'm happy to provide guidance and review code.

**Comment by jfy133** [comment]
Agreed! A sourmash subwork was actually already started on the taxprofiler repo (it's in a draft state at the moment), but the person taking that on seems to have not been able to finish it. On 'our side's we normally we add tools to taxpasta once it's in the pipeline as then we know exactly what is available etc.

That said I'm also happy to guide on the taxprofiler/nextflow side of things (I'm still on half tjme parental leave until August) , if someone wants to take over the half done subworkflow! We have a profiler -contribution guide for that too 

And agreed sylph also looks very interesting 👍
